src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  decreaseHealth(value) {
+    this.health -= value;
+
+    if (this.health <= 0) {
+      this.removeDeadAnimal();
+    }
+  }
+
+  removeDeadAnimal() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
```

Comment:
Add constructor method.

Reply:
Where and why? I don't need constructor method at Carnivore class.